test/models/Offer/OfferApprovalEvent/Repository.test.ts

```diff
@@ -0,0 +1,108 @@
+      const adminUserUuid = admin.userUuid;
+      const event = await OfferApprovalEventRepository.create({
+        adminUserUuid,
+        offer,
+        status
+      });
+      expect(event.adminUserUuid).toEqual(admin.userUuid);
+      expect(event.offerUuid).toEqual(offer.uuid);
+      expect(event.status).toEqual(status);
+    };
+
+    it("creates a valid OfferApprovalEvent with approved status", async () => {
+      await expectValidCreationWithStatus(ApprovalStatus.approved);
+    });
+
+    it("creates a valid OfferApprovalEvent with rejected status", async () => {
+      await expectValidCreationWithStatus(ApprovalStatus.rejected);
+    });
+
+    it("creates a valid OfferApprovalEvent with pending status", async () => {
+      await expectValidCreationWithStatus(ApprovalStatus.pending);
+    });
+
+    it("throws an error if adminUserUuid does not belong to an admin", async () => {
+      const company = await CompanyGenerator.instance.withCompleteData();
+      const notAnAdminUserUuid = "b11a1432-1a79-4bce-aa2b-9dd0cb6aa648";
+
+      const offer = await OfferGenerator.instance.withObligatoryData({ companyUuid: company.uuid });
+      const status = ApprovalStatus.approved;
+      await expect(
+        OfferApprovalEventRepository.create({
+          adminUserUuid: notAnAdminUserUuid,
+          offer,
+          status
+        })
+      ).rejects.toThrowErrorWithMessage(
+        ForeignKeyConstraintError,
+        'insert or update on table "OfferApprovalEvents" violates ' +
+          'foreign key constraint "OfferApprovalEvents_adminUserUuid_fkey"'
+      );
+    });
+
+    it("gets offer and admin by association", async () => {
+      const company = await CompanyGenerator.instance.withCompleteData();
+      const admin = await AdminGenerator.instance({ secretary: Secretary.extension });
+      const status = ApprovalStatus.approved;
+      const adminUserUuid = admin.userUuid;
+      const offer = await OfferGenerator.instance.withObligatoryData({ companyUuid: company.uuid });
+      const event = await OfferApprovalEventRepository.create({
+        adminUserUuid,
+        offer,
+        status
+      });
+
+      expect((await event.getOffer()).toJSON()).toEqual(offer.toJSON());
+      expect((await event.getAdmin()).toJSON()).toEqual(admin.toJSON());
+    });
+  });
+
+  describe("Delete cascade", () => {
+    const createOfferApprovalEvent = async () => {
+      const company = await CompanyGenerator.instance.withCompleteData();
+      const { userUuid: adminUserUuid } = await AdminGenerator.instance({
+        secretary: Secretary.extension
+      });
+      const status = ApprovalStatus.approved;
+      const offer = await OfferGenerator.instance.withObligatoryData({ companyUuid: company.uuid });
+      return OfferApprovalEventRepository.create({
+        adminUserUuid,
+        offer,
+        status
+      });
+    };
+
+    it("deletes all events if offer table is truncated", async () => {
```

Comment:
una boludez, pero aca offers en plural:
`deletes all events if offers table is truncated`